hydromt/model/components/geoms.py

```diff
@@ -171,12 +173,19 @@ def write(
             if None, the path that was provided at init will be used.
         to_wgs84: bool, optional
             If True, the geoms will be reprojected to WGS84(EPSG:4326) before they are written.
+        region_options: dict, optional
+            Options to pass to the write_region method.
+            Can contain `filename`, `to_wgs84`, and anything that will be passed to `GeoDataFrame.to_file`.
+            If `filename` is not provided, `self.region_filename` will be used.
         **kwargs:
             Additional keyword arguments that are passed to the
             `geopandas.to_file` function.
         """
         self.root._assert_write_mode()
 
+        region_options = region_options or {}
+        self.write_region(**region_options)
```

Comment:
if None can we just skip writting the region altogether instead of empty dict and still writting? I think I would like None to be the default as for most components I think you do not want this. Ie for wflow only for gridcomponent we want that and not for all others